matRad/dicom/@matRad_DicomImporter/matRad_scanDicomImportFolder.m

```diff
@@ -0,0 +1,307 @@
+            matRad_progress(numOfFiles+1-i, numOfFiles);
+        end
+        
+    end
+
+    % Filtration, getting and assigning z resolution to all CT files
+    FiltredLocArray = unique(LocationsArray);
+    Thickness = rmmissing(unique(diff(FiltredLocArray)));
+    numOfFiles = numel(obj.allfiles(:,1));
+   
+    if numel(Thickness) > 1
+        matRad_cfg.dispWarning('Slices are not equidistant! CT will be interpolate with the lowest value of resolution is given by CT slices.');
+        Thickness = Thickness(1); % min thickness
+        ThBool = []; % in this case we will also create ct cube with the same resolution for all slices 
+        PriorThicknesses = flip(rmmissing(diff(FiltredLocArray))); % prior values of spacing, which are needed for interpolation
+        Counter = 0;
+        
+        for i = 1:numOfFiles
+            if strcmp(obj.allfiles{i,2},'CT') && (i - Counter <= numel(PriorThicknesses))
+                obj.allfiles{i,16} = num2str(PriorThicknesses(i - Counter));
+            else 
+                Counter = Counter + 1;
+                obj.allfiles{i,16} = NaN;
+            end
+        end
+        
+    end
+
+    if isempty(ThBool)
+        for i = numOfFiles:-1:1
+            if strcmp(obj.allfiles{i,2},'CT') 
+                obj.allfiles{i,11} = num2str(Thickness);
+            end
+        end
+    end    
+    close(h)
+    
+    if ~isempty(obj.allfiles)
+        obj.patient = unique(obj.allfiles(:,3));
+        
+        if isempty(obj.patient)
+             matRad_cfg.dispError('No patient found with DICOM CT _and_ RT structure set in patient directory!');
+        end
+    else
+         matRad_cfg.dispError('No DICOM files found in patient directory!');
+        %h.WindowStyle = 'Modal';
+        %error('No DICOM files found in patient directory');
+    end
+else
+    matRad_cfg.dispError('Folder is empty!');
+end
+
+clear warnDlgDICOMtagShown;
+
+end
+
+function allfiles = parseDicomTag(allfiles,info,tag,row,column,parsefcn)
+
+global warnDlgDICOMtagShown;
+
+defaultPlaceHolder = '001';
+
+if nargin < 6
+    parsefcn = @(x) x;
+end
+
+try
+   if isfield(info,tag)
+      if ~isempty(info.(tag))
+         allfiles{row,column} = parsefcn(info.(tag));
+      else
+         allfiles{row,column} = defaultPlaceHolder;
+      end
+   else
+      allfiles{row,column} = defaultPlaceHolder;
+   end
+catch
+   allfiles{row,column} = NaN;
+end
+
+if ~warnDlgDICOMtagShown && strcmp(allfiles{row,column},defaultPlaceHolder) && (column == 3 || column == 4)
+ 
+   dlgTitle    = 'Dicom Tag import';
+   dlgQuestion = ['matRad_scanDicomImportFolder: Could not parse dicom tag: ' tag '. Using placeholder ' defaultPlaceHolder ' instead. Please check imported data carefully! Do you want to continue?'];
+   answer      = questdlg(dlgQuestion,dlgTitle,'Yes','No', 'Yes');
+   
+   warnDlgDICOMtagShown = true;
+   
+   switch answer
+      case 'No'
+         matRad_cfg.dispError('Inconsistency in DICOM tags')  
+   end
+end
```

Comment:
This is opening a dialog waiting for an answer. Dialogs should not be opened in the functions (which might also beused from scripts) but in the corresponding GUI piece.